src/qtt/data.py

```diff
@@ -1029,183 +1031,326 @@ def saveExperimentData(outputdir, dataset, tag, dstr):
-        yname (str or list): the name of the measured array
-        y (array): the measured data
-    '''
-    xx = np.array(x)
-    yy = np.NaN * np.ones(xx.size) if y is None else np.array(y)
-    x = DataArray(name=xname, array_id=xname, preset_data=xx, unit=xunit, is_setpoint=True)
-    dd = new_data(arrays=(), location=location, loc_record=loc_record)
-    dd.add_array(x)
-    if isinstance(yname, str):
-        y = DataArray(name=yname, array_id=yname, preset_data=yy, unit=yunit, set_arrays=(x,))
-        dd.add_array(y)
+        p (qcodes.SweepFixedValues): the parameter to check
+
+    Raises:
+        TypeError: When p does not have attribute 'parameter'
+        TypeError: When attribute p.parameter is not an instance of qcodes.parameter.
+    """
+    if hasattr(p, 'parameter'):
+        if not isinstance(p.parameter, qcodes.Parameter):
+            raise TypeError('Type of p.parameter must be qcodes.Parameter')
     else:
-        for ii, name in enumerate(yname):
-            y = DataArray(name=name, array_id=name, preset_data=yy[ii], unit=yunit, set_arrays=(x,))
-            dd.add_array(y)
-    return dd
+        raise TypeError('Type of p must be qcodes.SweepFixedValues')
 
 
-def makeDataSet1D(x, yname='measured', y=None, location=None, loc_record=None, return_names=False):
-    ''' Make DataSet with one or multiple 1D arrays and one setpoint array.
+def _make_data_set(measured_data_list, measurement_list, measurement_unit, location, loc_record, preset_data,
+                   setpoints):
+    """ Generic code to make the data set for makeDataSet1D, makeDataSet1DPlain, makeDataSet2D, makeDataSet2DPlain
 
-    Arguments:
-        x (array): the setpoint array of data
-        yname (str or list): name(s) of measured array(s)
-        y (array or None): optional array to fill the DataSet
-        location (str or None): location for the DataSet
-        loc_record (dict): will be added to the location
-        return_names (bool): if True return array names in output
-    '''
-    xx = np.array(x)
-    yy = np.NaN * np.ones(xx.size)
-    x = DataArray(name=x.name, array_id=x.name, label=x.parameter.label,
-                  unit=x.parameter.unit, preset_data=xx, is_setpoint=True)
-    if isinstance(yname, str):
-        measure_names = [yname]
-        if y is not None:
-            preset_data = [y]
+    Raises:
+        ValueError: When the number of measurements names in the measurement_list doesn not match the number
+                    of measurements.
+
+        ValueError: When the shape of the measured data is not matching the shape of the setpoint array. When the
+                    measured data is a list, each list item must match the shape of the setpoint array.
+
+        TypeError: When a measurement name in the measurement list has an invalid type.
+
+    Returns:
+        The resulting data set and the measure names list
+    """
+    dd = new_data(arrays=(), location=location, loc_record=loc_record)
```

Comment:
Can we change the variable `dd` to something more understandable?

Reply:
Yup, data_set it is...